ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml

```diff
@@ -78,6 +101,244 @@ tests:
+        for i in $(seq ${RETRIES}); do
+          if [[ $(oc get vm -n ${CLUSTER_NS} --no-headers | wc -l) -lt 2 ]]; then
+            sleep 30
+            echo "Try ${i}/${RETRIES}: not all the VMs were created, yet; trying again in 30 seconds"
+            continue
+          else
+            VMS_EXISTS=true
+            break
+          fi
+        done
+
+        if [[ ${VMS_EXISTS} == "false" ]]; then
+          echo "should create at least 2 VMs, but it didn't"
+          exit 1
+        fi
+
+        #check VMs
+        for VM in $(oc get vm -n ${CLUSTER_NS} -o custom-columns=NAME:.metadata.name --no-headers); do
+          oc wait vm -n ${CLUSTER_NS} ${VM} --for=condition=Ready
+        done
+
+        #check VMIs
+        for VMI in $(oc get vmi -n ${CLUSTER_NS} -o custom-columns=NAME:.metadata.name --no-headers); do
+          oc wait vmi -n ${CLUSTER_NS} ${VMI} --for=condition=Ready
+          [[ $(oc get vmi -n ${CLUSTER_NS} ${VMI} -o custom-columns=PHASE:.status.phase --no-headers) == "Running" ]]
+        done
+
+        oc get vmi -n ${CLUSTER_NS}
+
+        TENANT_CLUSTER_IP=$(oc get svc -n ${CLUSTER_NS} -l cluster.x-k8s.io/cluster-name=tenant-capk -o custom-columns=EXT_IP:.status.loadBalancer.ingress[0].ip --no-headers)
+        clusterctl get kubeconfig ${CLUSTER_NAME} -n ${CLUSTER_NS} > ${CLUSTER_NAME}-kubeconfig
+        sed -i -E "/certificate-authority-data/d;s|(server: https://)[^:]+|\1${TENANT_CLUSTER_IP}|;/^- cluster:/a\    insecure-skip-tls-verify: true" ${CLUSTER_NAME}-kubeconfig
+
+        # check the tenant cluster. Check if the API-Server is responsive
+        MNGMNT_KUBECONFI=${KUBECONFIG}
+        export KUBECONFIG=${CLUSTER_NAME}-kubeconfig
+
+        RETRIES=20
+        API_SERVER_IS_UP=false
+        for i in $(seq ${RETRIES}); do
+          if [[ $(oc get --raw='/readyz') ]]; then
+            API_SERVER_IS_UP=true
+            break
+          else
+            echo "Try ${i}/${RETRIES}: tenant API Server is not up, yet; trying again in 30 seconds"
+            sleep 30
+            continue
+          fi
+        done
+
+        if [[ ${API_SERVER_IS_UP} == "false" ]]; then
+          echo "Should have a responsive tenenat API server by now; aborting..."
+          exit 1
+        fi
+
+        RETRIES=20
+        NODES_ARE_UP=false
+        for i in $(seq ${RETRIES}); do
+          if [[ $(oc get nodes --no-headers | wc -l) -lt 2 ]]; then
+            echo "Try ${i}/${RETRIES}: not all the nodes are up, yet; trying again in 30 seconds"
+            sleep 30
+            continue
+          else
+            NODES_ARE_UP=true
+            break
+          fi
+        done
+
+        if [[ ${NODES_ARE_UP} == "false" ]]; then
+          echo "Should have two working nodes by now; aborting..."
+          exit 1
+        fi
+
+        oc get nodes -o wide
+
+        oc config set-context $(oc config current-context) --namespace=default
+
+        # create a workload on the tenant cluster
```

Comment:
I think this shouldn't be done here, maybe better to add the test in capk repository, also to be used upstream?
Need to be executed in another step